include/nbl/ui/CWindowManagerXCB.h

```diff
@@ -0,0 +1,94 @@
+#ifndef C_WINDOW_MANAGER_XCB
+#define C_WINDOW_MANAGER_XCB
+
+#ifdef _NBL_PLATFORM_LINUX_
+#include "nbl/core/decl/Types.h"
+
+#include "nbl/system/DefaultFuncPtrLoader.h"
+
+#include "nbl/ui/IWindow.h"
+#include "nbl/ui/IWindowManager.h"
+
+#include <functional>
+#include <memory>
+#include <string>
+
+#include <xcb/xcb.h>
+#include <xcb/xcb_icccm.h>
+#include <xcb/xproto.h>
```

Comment:
it would be decent to follow the Win32 pattern in #476 and create a `IWindowManagerXCB` as a public header with a single method
```cpp
static core::smart_refctd_ptr<IWindowManagerXCB> create();
```
then hide all these implementation details in `src`